### PR TITLE
DOCS-#2963: document modin.experimental.pandas

### LINKED ISSUE #2963: Document modin.experimental.pandas
This should include docstrings and purposes of everything inside `modin.experimental.pandas` module.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [x] Resolves #2963 
- [ ] tests added and passing

### CODE REVIEW

File: modin/experimental/engines/pandas_on_ray/sql.py [hunk 1]
```diff
@@ -184,6 +184,7 @@ def get_query_info(sql, con, partition_column):
         cols = get_query_columns(engine, query)
     # TODO allow validation that takes into account edge cases of pandas e.g. "[index]"
     # check_partition_column(partition_column, cols)
+    # partition_column isn't used
```
Comment (vnlitvinov) [inline modin/experimental/engines/pandas_on_ray/sql.py:187]: This should be a `FIXME` or smth comment (with a call-to-action) describing what to do

File: modin/experimental/pandas/__init__.py [hunk 2]
```diff
@@ -11,6 +11,26 @@
 # ANY KIND, either express or implied. See the License for the specific language
 # governing permissions and limitations under the License.
 
+"""
+The main module through which interaction with the experimental API takes place.
+
+See `Experimental API Ireference` for details.
```
Comment (vnlitvinov) [inline modin/experimental/pandas/__init__.py:17]:
```suggestion
See `Experimental API Reference` for details.
```
We might want to include a direct link to docs here, as this docstring is used _both_ to generate a docsite and to be viewed locally, e.g. in an IDE

File: modin/experimental/pandas/io_exp.py [hunk 3]
```diff
@@ -32,41 +34,64 @@ def read_sql(
...
-
-    Returns:
-        Pandas Dataframe
+    partition_column: Optional[str] = None,
+    lower_bound: Optional[int] = None,
+    upper_bound: Optional[int] = None,
+    max_sessions: Optional[int] = None,
+) -> DataFrame:
+    """
+    General documentation in `modin.pandas.read_sql`.
```
Comment (vnlitvinov) [inline modin/experimental/pandas/io_exp.py:43]:
```suggestion
    General documentation is available in `modin.pandas.read_sql`.
```

File: modin/experimental/pandas/io_exp.py [hunk 4]
```diff
@@ -32,41 +34,64 @@ def read_sql(
...
+
+    Experimental feature is simultaneous reading from a sql file.
```
Comment (vnlitvinov) [inline modin/experimental/pandas/io_exp.py:45]:
```suggestion
    This experimental feature provides distributed reading from a sql file.
```

File: modin/experimental/pandas/io_exp.py [hunk 5]
```diff
@@ -32,41 +34,64 @@ def read_sql(
...
+    lower_bound : int, optional
+        The minimum value to be requested from the partition_column.
+    upper_bound : int, optional
+        The maximum value to be requested from the partition_column.
+    max_sessions : int, optional
+        The maximum number of simultaneous connections allowed to use.
+
+    Returns
+    -------
+    Modin DataFrame.
```
Comment (vnlitvinov) [inline modin/experimental/pandas/io_exp.py:94]:
```suggestion
    modin.DataFrame
```
there should be return type (which might have missing description)

File: modin/experimental/pandas/io_exp.py [hunk 6]
```diff
@@ -154,16 +179,21 @@ def parser_func(
     return parser_func
 
 
-def _read(**kwargs):
+def _read(**kwargs) -> DataFrame:
     """
-    Read csv file from local disk.
+    General documentation in `modin.pandas.read_csv`.
```
Comment (vnlitvinov) [inline modin/experimental/pandas/io_exp.py:184]:
```suggestion
    General documentation is available in `modin.pandas.read_csv`.
```

File: modin/experimental/pandas/io_exp.py [hunk 7]
```diff
@@ -154,16 +179,21 @@ def parser_func(
...
+
+    Experimental feature is simultaneous reading from multiple csv files which are
+    defined using glob pattern. Works only for local files.
```
Comment (vnlitvinov) [inline modin/experimental/pandas/io_exp.py:187]:
```suggestion
    This experimental feature provides parallel reading from multiple csv files which are
    defined by glob pattern. Works for local files only!
```

File: modin/experimental/pandas/io_exp.py [hunk 8]
```diff
@@ -154,16 +179,21 @@ def parser_func(
...
-    filepath_or_buffer:
-        The filepath of the csv file.
-        We only support local files for now.
-    kwargs: Keyword arguments in pandas.read_csv
+    **kwargs : dict
+        Keyword arguments in `modin.pandas.read_csv`.
+
+    Returns
+    -------
+    Modin DataFrame.
```
Comment (vnlitvinov) [inline modin/experimental/pandas/io_exp.py:196]:
```suggestion
    modin.DataFrame
```
this should be a type, not description

File: modin/experimental/pandas/io_exp.py [hunk 9]
```diff
@@ -186,3 +216,4 @@ def _read(**kwargs):
 
 
 read_csv_glob = _make_parser_func(sep=",")
+read_csv_glob.__doc__ = _read.__doc__
```
Comment (vnlitvinov) [inline modin/experimental/pandas/io_exp.py:219]: I suggest moving this line to `_make_parser_func()` itself

File: modin/experimental/pandas/numpy_wrap.py [hunk 10]
```diff
@@ -45,11 +47,26 @@
...
+        Replacement occurs by getting attributes from either local NumPy or remote one when remote context
         is activated.
-        It also registers helpers for pickling local numpy objects in remote context
+        It also registers helpers for pickling local NumPy objects in remote context
         and vice versa.
+
+        Attributes
+        ----------
+        __own_attrs__ : set
+            Attributes that defined in the class.
```
Comment (vnlitvinov) [inline modin/experimental/pandas/numpy_wrap.py:60]:
```suggestion
            Attributes that are defined in this class so access to them must never be proxied.
```

File: modin/experimental/pandas/numpy_wrap.py [hunk 11]
```diff
@@ -45,11 +47,26 @@
...
+        __current_numpy : ModuleType
+            The module to which getting NumPy attributes redirects. For example,
+            NumPy in remote machine.
```
Comment (vnlitvinov) [inline modin/experimental/pandas/numpy_wrap.py:63]:
```suggestion
            NumPy on remote machine.
```

**Comment by anmyachev** [comment]
@vnlitvinov ready for review

File: docs/flow/modin/experimental/pandas.rst [hunk 12]
```diff
@@ -0,0 +1,13 @@
+:orphan:
+
+Experimental Pandas Module Description
```
Comment (devin-petersohn) [inline docs/flow/modin/experimental/pandas.rst:3]:
```suggestion
Experimental Pandas API
```

File: modin/experimental/engines/pandas_on_ray/sql.py [hunk 13]
```diff
@@ -184,6 +184,8 @@ def get_query_info(sql, con, partition_column):
         cols = get_query_columns(engine, query)
     # TODO allow validation that takes into account edge cases of pandas e.g. "[index]"
     # check_partition_column(partition_column, cols)
+    # TODO partition_column isn't used; we need to either use or remove it;
```
Comment (devin-petersohn) [inline modin/experimental/engines/pandas_on_ray/sql.py:187]: Do we have an issue open for this? It should be used to determine the partitioning.

**Review by devin-petersohn — COMMENTED** [review]
Looks good, since this is API docs I prefer the minor renaming. Also left a quesiton

See build here:
https://modin--2964.org.readthedocs.build/en/2964/flow/modin/experimental/pandas.html

File: modin/experimental/engines/pandas_on_ray/sql.py (hunk 13 above)
Comment (anmyachev) [reply modin/experimental/engines/pandas_on_ray/sql.py:187]: Created - https://github.com/modin-project/modin/issues/3050

**Review by devin-petersohn — COMMENTED** [review]
@anmyachev I think this is good to merge, we just need it to be rebased.

**Comment by anmyachev** [comment]
> @anmyachev I think this is good to merge, we just need it to be rebased.

@devin-petersohn rebased

**Review by devin-petersohn — APPROVED** [review]
Thanks @anmyachev, LGTM!

**Pull request merged**